cmd/manager/main.go

```diff
@@ -130,22 +145,33 @@ func main() {
 	dependentSelector := k8slabels.NewSelector().Add(*dependentRequirement)
 
 	setupLog.Info("set up manager")
+	cacheOptions := crcache.Options{
+		ByObject: map[client.Object]crcache.ByObject{
+			&ocv1alpha1.ClusterExtension{}: {Label: k8slabels.Everything()},
+			&catalogd.ClusterCatalog{}:     {Label: k8slabels.Everything()},
+		},
+		DefaultNamespaces: map[string]crcache.Config{
+			systemNamespace: {LabelSelector: k8slabels.Everything()},
+		},
+		DefaultLabelSelector: dependentSelector,
+	}
+	if globalPullSecretKey != nil {
+		cacheOptions.ByObject[&corev1.Secret{}] = crcache.ByObject{
+			Field: fields.SelectorFromSet(map[string]string{
+				"metadata.name": globalPullSecretKey.Name,
+			}),
+		}
+		cacheOptions.DefaultNamespaces[globalPullSecretKey.Namespace] = crcache.Config{
+			LabelSelector: k8slabels.Everything(),
+		}
```

Comment:
This doesn't seem quite right to me, but we may need to experiment to know for sure. I feel like the following should be sufficient (but maybe it isn't?):
```suggestion
		cacheOptions.ByObject[&corev1.Secret{}] = crcache.ByObject{
			Namespaces: map[string]crcache.Config{
				globalPullSecretKey.Namespace: crcache.Config{
					Field: fields.SelectorFromSet(map[string]string{
						"metadata.name": globalPullSecretKey.Name,
					},
				},	
				systemNamespace: crcache.Config{
					LabelSelector: k8slabels.Everything(),
				},
			}
		}
```

Reply:
Nah that didn't work, what I have is the only combination that's worked so far.

Reply:
Also in case you missed it: [comment](https://github.com/operator-framework/operator-controller/pull/1322#discussion_r1786044859)

Reply:
We got this sorted out. I pushed a new commit with improved cache options.